.github/workflows/rust.yml

```diff
@@ -26,11 +26,11 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y libudev-dev
     - name: Build (Default)
-      run: cargo build
-    - name: Build (All features)
-      run: cargo build --all-features
+      run: cargo clean && cargo build
```

Comment:
I don't think this is actually necessary, and I can't reproduce any build failure.

Reply:
```
$ git fetch origin && git checkout main
Already on 'main'
Your branch is up to date with 'origin/main'.
$ cargo clean && cargo build -q
warning: unused variable: `clk_data`
   --> rinex/src/qc/analysis/obs.rs:309:17
    |
309 |             let clk_data = rec.filter(mask);
    |                 ^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_clk_data`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
   --> rinex/src/qc/analysis/obs.rs:302:13
    |
302 |         let mut clock_drift: Option<f64> = None;
    |             ----^^^^^^^^^^^
    |             |
    |             help: remove this `mut`
    |
    = note: `#[warn(unused_mut)]` on by default

warning: field `clock_diff` is never read
  --> rinex/src/hatanaka/compressor.rs:43:5
   |
23 | pub struct Compressor {
   |            ---------- field in this struct
...
43 |     clock_diff: NumDiff,
   |     ^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: field `clock_drift` is never read
   --> rinex/src/qc/analysis/obs.rs:280:5
    |
255 | pub struct QcObsAnalysis {
    |            ------------- field in this struct
...
280 |     clock_drift: Option<f64>,
    |     ^^^^^^^^^^^
    |
    = note: `QcObsAnalysis` has derived impls for the traits `Clone` and `Debug`, but these are intentionally ignored during dead code analysis

warning: use of deprecated method `chrono::NaiveDate::and_hms`: use `and_hms_opt()` instead
  --> sinex/src/datetime.rs:20:11
   |
20 |     Ok(dt.and_hms(h as u32, m as u32, s as u32))
   |           ^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default

$
```

Reply:
So it passed.. this is confusing.

When building with no features, "processing" feature is disabled.
But `src/ionex/record.rs` tries to import processing features whether it is enabled or not

Reply:
Ah, I get the failure with `cargo build --all-targets`.

Reply:
Something happens when you invoke cargo twice, with different options.

Reply:
It's because `rinex-cli` enables the `processing` feature. `cargo build -p rinex` doesn't (and neither does `cargo package`, I suppose), so you get the compilation errors. I'd use `--all-targets`, but it currently doesn't work because of some code that's now missing.

Reply:
@lnicola 

>  I'd use --all-targets

The problem with this, is it seems to end up the tests without any flags.
And we need at least `--feature tests` to be passed when running these methods,
to include the test methods. I have grown tiered to write and rewrite the same thing all the time, and it makes the test infra much more efficient to share methods.

Unfortunately I have not found any clean methods to include shared methods into different "tests".  
The only way is to put it in the library directly.

Reply:
Okay, plan B is to run `cargo build -p` on each crate :-(.

Reply:
Can you gate `test_toolkit` on `cfg(test)` instead?

Reply:
that's a good idea. I'll give it a try.

How can I make the "requested changed" as solved ?

Reply:
> Can you gate `test_toolkit` on `cfg(test)` instead?

that is already the case, see the first few lines, for rinex/src/lib.rs

Reply:
I don't think I see it, can you link? I meant cfg(test), not the tests feature.